test/test_write.ml

```diff
@@ -29,7 +29,24 @@ let to_file () =
   test ~newline:true ();
   test ~newline:false ()
 
+let stream_to_file () =
+  let output_file = Filename.temp_file "test_yojson_stream_to_file" ".json" in
+  let data = [`String "foo"; `String "bar"] in
+  Yojson.Safe.stream_to_file output_file (Stream.of_list data);
+  let read_data =
+    let stream = Yojson.Safe.stream_from_file output_file in
+    let acc = ref [] in
+    Stream.iter (fun v -> acc := v :: !acc) stream;
+    List.rev !acc
+  in
+  Sys.remove output_file;
+  if data <> read_data then
+    (* TODO: it would be nice to use Alcotest.check,
```

Comment:
If you want you can add an instance for `Yojson.Safe.t` if you want, it would be a nice improvement and should be quite simple to do (and actually sensible in the long run, I think the reason why there isn't one is because the tests are mostly translated from OUnit. But this is up for you, a nice to have but not necessary.